doc/setup/ursim_docker.rst

```diff
@@ -0,0 +1,126 @@
+Docker containers using your current user.
+
+Start a URSim docker container
+------------------------------
+
+To startup a simulated robot run the following command. This will start a Docker container named
+``ursim`` and startup a simulated UR5e robot. It exposes ports 5900 and 6080 for the browser-based
+polyscope access. Note that this will expose the simulated robot to your local area network if you
+don't have any further level of security such as a firewall active. To prevent this, you can either
+skip the port forwarding instructions (skip the two ``-p port:port`` statements) in which case
+you'll have to use the container's IP address to access the polyscope gui rather than ``localhost`` or
+you can restrict the port forwarding to a certain network interface (such as the looppack interface)
+see Docker's upstream documentation on port exposure for further information.
+
+.. code-block:: bash
+
+   docker run --rm -it -p 5900:5900 -p 6080:6080 --name ursim universalrobots/ursim_e-series
+
+External Control
+----------------
+
+To use the external control functionality, we will need the ``external_control`` URCap installed on
+the robot and a program containing its *ExternalControl* program node. Both can be prepared on the
+host machine either by creating an own Dockerfile containing those or by mounting two folders
+containing installed URCaps and programs. See the Dockerfile's upstream `documentation <https://hub.docker.com/r/universalrobots/ursim_e-series>`_.
+
+In this example, we will bind-mount a folder for the programs and URCaps. First, let's create a
+local folder where we can store things inside:
+
+.. code-block:: bash
+
+   mkdir -p ${HOME}/.ursim/programs
+   mkdir -p ${HOME}/.ursim/urcaps
+
+Then, we can "install" the URCap by placing its ``.jar`` file inside the urcaps folder
+
+.. code-block:: bash
+
+   URCAP_VERSION=1.0.5 # latest version as if writing this
+   curl -L -o ${HOME}/.ursim/urcaps/externalcontrol-${URCAP_VERSION}.jar \
+     https://github.com/UniversalRobots/Universal_Robots_ExternalControl_URCap/releases/download/v${URCAP_VERSION}/externalcontrol-${URCAP_VERSION}.jar
+
+With this, start your URSim containers with the following command:
+
+.. code-block:: bash
+
+   docker run --rm -it -p 5900:5900 -p 6080:6080 -v ${HOME}/.ursim/urcaps:/urcaps -v ${HOME}/.ursim/programs:/ursim/programs --name ursim universalrobots/ursim_e-series
+
+With this, you should be able to setup the ``external_control`` URCap and create a program as
+described in :ref:`URCap setup guide <install_urcap>`.
+
+Network setup
+-------------
+
+As described above, you can always start the URSim container using the default network setup. As long
+as you don't have any other docker containers running, it will most probably always get the same IP
+address assigned every time. However, to make things a bit more explicit, we can setup our own
+docker network where we can assign a static IP address to our URSim container.
```

Comment:
```suggestion
As described above, you can always start the URSim container using the default network setup. It will most probably
always get the same IP address assigned every time, as long as you don't have any other docker containers running.
However, to make things a bit more explicit, we can setup our own docker network where we can assign a static IP
address to our URSim container.
```